### DISCUSSION TITLE
Switching from Swipeable to Swipeable Reanimated > ref > Swipeable vs SwipeableMethods

### DISCUSSION THREAD

**Discussion by markaustinws** [software-mansion/react-native-gesture-handler#3051] [Q&A]
Hello,
I've noticed that the original Swipeable component is being depreicated and being replaced with Swipeable Reanimated.

There is a breaking change for me.  I was using the state object from Swipeable to detect the length of swipe.  Our ask is for the Swipeable to act differently on a long swipe as opposed to a short swipe.

I have tried wrapping a Pan gesture around the Swipeable Reanimated component to cover this.  However, it seems to be conflicting wit h the swipe actions in the Swipeable Reanimated component.

Is it a possible to revert SwipeableMethods to Swipeable as a reference?  Or is there another way of detecting swipe length?

Thanks in advance

Mark

**Comment by latekvo** [comment]
Hey @markaustinws 

Functions passed to `renderLeftActions` and `renderRightActions` are provided with the following arguments:
- `progress` - value representing swiping progress relative to the width of the returned element. From `0` to `1`.
- `translation` - horizontal translation of `swipeable` in pixels.

I believe these are the values you're looking for.

I'm aware the documentation might've been unclear, i proposed some changes to the part mentioning `translation` and `progress` monitoring in [this PR](https://github.com/software-mansion/react-native-gesture-handler/pull/3121), please let me know if they're easier to understand, and if you have any suggestions as to how they could be improved, also please let us know.
